p2p/src/network/yamux/p2p_network_yamux_reducer.rs

```diff
@@ -211,11 +214,37 @@ impl P2pNetworkYamuxState {
                             }
                         }
                         YamuxFrameInner::WindowUpdate { difference } => {
-                            yamux_state
+                            let stream = yamux_state
                                 .streams
                                 .entry(frame.stream_id)
-                                .or_insert_with(YamuxStreamState::incoming)
-                                .update_window(false, difference);
+                                .or_insert_with(YamuxStreamState::incoming);
+                            stream.update_window(false, difference);
+                            if difference > 0 {
```

Comment:
this can be 0 or negative? in which cases does that happen?

Reply:
ok, please add a `bug_condition!` call in the else branch so that if this ever happens for some reason we will know about it. That is something we want to do in general, to avoid such unexpected (unenforced) cases happening silently without us noticing

Reply:
I add a runtime error, not bug_condition, because peer can send us this and we should not crash.

Reply:
You are right, this is based on external input, so it is not exactly a `bug_condition!`. Btw `bug_condition` only panics if a flag is enabled in the environment, otherwise it just logs an error (but again this specific case is not for it)